plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/adapter/MinishiftPoller.java

```diff
@@ -0,0 +1,145 @@
+/******************************************************************************* 
+ * Copyright (c) 2015 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/ 
+package org.jboss.tools.openshift.cdk.server.core.internal.adapter;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.wst.server.core.IServer;
+import org.jboss.ide.eclipse.as.core.JBossServerCorePlugin;
+import org.jboss.tools.openshift.cdk.server.core.internal.CDKConstants;
+import org.jboss.tools.openshift.cdk.server.core.internal.CDKCoreActivator;
+import org.jboss.tools.openshift.cdk.server.core.internal.MinishiftBinaryUtility;
+import org.jboss.tools.openshift.cdk.server.core.internal.VagrantBinaryUtility;
+import org.jboss.tools.openshift.cdk.server.core.internal.adapter.controllers.VagrantLaunchUtility;
+import org.jboss.tools.openshift.cdk.server.core.internal.adapter.controllers.VagrantTimeoutException;
+import org.jboss.tools.openshift.cdk.server.core.internal.listeners.CDKLaunchEnvironmentUtil;
+import org.jboss.tools.openshift.cdk.server.core.internal.listeners.ServiceManagerEnvironment;
+import org.jboss.tools.openshift.cdk.server.core.internal.listeners.ServiceManagerEnvironmentLoader;
+import org.jboss.tools.openshift.core.LazySSLCertificateCallback;
+
+import com.openshift.restclient.ClientBuilder;
+import com.openshift.restclient.IClient;
+import com.openshift.restclient.ISSLCertificateCallback;
+import com.openshift.restclient.OpenShiftException;
+
+public class MinishiftPoller extends AbstractCDKPoller {
+
+	protected void launchThread() {
+		launchThread("CDK Minishift Poller");
+	}	
+
+	protected Map<String, String> createEnvironment(IServer server) {
+		return CDKLaunchEnvironmentUtil.createEnvironment(server);
+	}
+	
+	@Override
+	public int getTimeoutBehavior() {
+		return TIMEOUT_BEHAVIOR_FAIL;
+	}
+
+	
+	private File getWorkingDirectory(IServer s) throws PollingException {
+		return JBossServerCorePlugin.getServerStateLocation(s).toFile();
+	}
+		
+	
+	protected IStatus onePing(IServer server, Map<String, String> env)
+			throws PollingException, IOException, TimeoutException {
+
+		String[] args = new String[] { CDKConstants.VAGRANT_CMD_STATUS };
```

Comment:
I think there is lots of code refereing to Vagrant concept/naming that need to be renamed in order for this code to be readable and maintainable